pkg/server/status_local_file_retrieval.go

```diff
@@ -97,6 +118,34 @@ func stacksLocal(req *serverpb.StacksRequest) (*serverpb.JSONResponse, error) {
 	return &serverpb.JSONResponse{Data: buf.Bytes()}, nil
 }
 
+// FilterStacksWithLabels writes all the stacks that have pprof labels matching
+// any one of the expectedLabels to resBuf.
+func FilterStacksWithLabels(stacks []byte, labelFilter string) []byte {
+	if labelFilter == "" {
+		return stacks
+	}
+	res := bytes.NewBuffer(nil)
+	goroutines := strings.Split(string(stacks), "\n\n")
+	// The first element is the nodeID which we want to always keep.
+	res.WriteString(goroutines[0])
+	goroutines = goroutines[1:]
+	for i, g := range goroutines {
+		// pprof.Lookup("goroutine") with debug=1 renders the pprof labels
+		// corresponding to a stack as:
+		// # labels: {"foo":"bar", "baz":"biz"}.
+		regex := regexp.MustCompile(fmt.Sprintf(`labels: {.*%s.*}`, labelFilter))
```

Comment:
Move this out to an init-type var. I know `regexp` has an internal cache but still!